pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -443,13 +330,171 @@ func (r *ReconcileComplianceScan) phaseAggregatingHandler(instance *complianceop
+		log.Error(err, "Cannot delete result server cert secret")
+		return reconcile.Result{}, err
+	}
+
+	if err = r.deleteResultClientSecret(instance, logger); err != nil {
+		log.Error(err, "Cannot delete result client cert secret")
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceScan) handleRootCASecret(instance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) error {
+	exist, err := secretExists(r.client, RootCAPrefix+instance.Name, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+	if exist {
+		return nil
+	}
+
+	logger.Info("creating CA", "instance", instance.Name)
+	secret, err := makeCASecret(instance, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+
+	// Create the CA secret.
+	err = r.client.Create(context.TODO(), secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	if err = controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileComplianceScan) handleResultServerSecret(instance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) error {
+	exist, err := secretExists(r.client, ServerCertPrefix+instance.Name, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+	if exist {
+		return nil
+	}
+
+	logger.Info("creating server cert", "instance", instance.Name)
+	secret, err := makeServerCertSecret(r.client, instance, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+
+	// Create the server cert secret.
+	err = r.client.Create(context.TODO(), secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	if err = controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileComplianceScan) handleResultClientSecret(instance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) error {
+	exist, err := secretExists(r.client, ClientCertPrefix+instance.Name, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+	if exist {
+		return nil
+	}
+
+	logger.Info("creating client cert", "instance", instance.Name)
+	secret, err := makeClientCertSecret(r.client, instance, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+
+	// Create the client cert secret.
+	err = r.client.Create(context.TODO(), secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	if err = controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileComplianceScan) deleteRootCASecret(instance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) error {
+	logger.Info("deleting CA", "instance", instance.Name)
+	secret, err := makeCASecret(instance, common.GetComplianceOperatorNamespace())
```

Comment:
These functions do all the key generation and stuff, it would be better to just get the names

Reply:
Right. It would have been better to have a function to get the names. I'll fix that.